sc/managers/configloaders/kubernetes/kubeconnector/connector.go

```diff
@@ -0,0 +1,179 @@
+package kubeconnector
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/spacecloud-io/space-cloud/config"
+	"github.com/spaceuptech/helpers"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/informers/internalinterfaces"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+)
+
+// Connector is use as connector for KubeStore module
+type Connector struct {
+	Lock sync.RWMutex
+
+	// Config related to cluster config
+	clusterID      string
+	ProjectsConfig *config.Config
+
+	kube    *kubernetes.Clientset
+	stopper chan struct{}
+}
+
+const spaceCloud string = "space-cloud"
+
+// New create a new instance of the Module object
+func New() (*Connector, error) {
+	c := &Connector{}
+
+	// Create the kubernetes client
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	kube, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set kube client and initialise stopper
+	c.kube = kube
+	c.stopper = make(chan struct{})
+
+	// Set cluster-id
+	c.clusterID = viper.GetString("cluster-id")
+
+	// Set global config
+	config, err := getGlobalConfig(c.kube, c.clusterID)
+	if err != nil {
+		return nil, err
+	}
+	c.ProjectsConfig = config
+
+	if err := c.getSCConfig(); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+// getSCConfig gets sc config from config map using shared informers
+func (c *Connector) getSCConfig() error {
+	// Start routine to observe space cloud project level resources
+	if err := watchResources(c.kube, c.clusterID, c.stopper, func(eventType, resourceID string, resourceType config.Resource, resource interface{}) {
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+
+		isSkip, err := c.validateResource(ctx, eventType, resourceID, resourceType, resource)
```

Comment:
We can completely remove the validate resource function. we dont need it anymore